### PR TITLE
List audits by ticket

### PR DESCRIPTION
**Pull request by FmeuMan**
Hi!
Just adding a new function to list Audits by ticket.
My goal is to find the tickets original email, looking for the first audit of email channel seems to do the trick.

### CODE REVIEW

**Comment by FmeuMan** [comment]
Thanks for your feedback. It should be ok by now.
Let me now if i should squash the commits

**Comment by eggman64** [comment]
> Thanks for your feedback. It should be ok by now.
> Let me now if i should squash the commits

Awesome! Thank you.

Could you perhaps add a test and then we can merge :ship: :)

File: zendesk/zendesk.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type Client interface {
 	ListOrganizationTickets(int64, *ListOptions, ...SideLoad) (*ListResponse, error)
 	ListExternalIDTickets(string, *ListOptions, ...SideLoad) (*ListResponse, error)
 	ListRequestedTickets(int64) ([]Ticket, error)
+	ListTicketAudits(ticketID int64, options *ListOptions) (*ListResponse, error)
```
Comment (eggman64) [inline zendesk/zendesk.go]: For consistency with the other method definitions, I recommend leaving out the variables `ticketID` and `options`

File: zendesk/zendesk.go [hunk 2]
```diff
@@ -267,6 +268,7 @@ func unmarshall(res *http.Response, out interface{}) error {
 type APIPayload struct {
 	Attachment                 *Attachment                `json:"attachment"`
 	Attachments                []Attachment               `json:"attachments"`
+	Audits                     []TicketAudit              `json:"audits"`
```
Comment (eggman64) [inline zendesk/zendesk.go]: Please `omitempty` here

**Comment by FmeuMan** [comment]
I added a test to list audits on a new ticket but I don't have access to a sandbox zendesk by now to run the tests :cry:
Listing audit on a production zendesk work thought

File: zendesk/audit_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	user := randUser(t, client)
+	defer client.DeleteUser(*user.ID)
+
+	ticket := randTicket(t, client, user)
+	defer client.DeleteTicket(*ticket.ID)
+
+	// assert that a newly created ticket has an audit
+	listed, err := client.ListTicketAudits(*ticket.ID, nil)
+	require.NoError(t, err)
+	require.Len(t, listed, 1)
```
Comment (eggman64) [inline zendesk/audit_test.go]: `listed.Audits` should be checked for length

**Comment by FmeuMan** [comment]
Ok test passed against a sandbox

**Pull request merged**